src/lib.rs

```diff
@@ -432,6 +443,12 @@ impl Clone for Variable {
     }
 }
 
+impl PartialEq for Variable {
```

Comment:
More descriptive function name? 
```suggestion
impl IdEq for Variable {
```

Reply:
Isn't PartialEq a Trait?